src/dmd/statementsem.d

```diff
@@ -2037,6 +2037,8 @@ else
         fs.gotos = new ScopeStatements();
         auto fld = new FuncLiteralDeclaration(fs.loc, fs.endloc, tf, TOK.delegate_, fs);
         fld.fbody = fs._body;
+        if (tfld && tfld.isShared())
+            fld.storage_class |= STC.shared_;
```

Comment:
Hmm, I think this is making a `shared (int delegate(...))` which by transitivity of shared makes the context pointer shared. what I need is for the context pointer to be shared, not the whole delegate.

Reply:
Ok, but now the error is related to the `-preview=nosharedaccess` switch, I've just checked that currently `core.atomic` is not prepared to be used with it.
Its unittest just gives a bunch of errors:
`dmd -unittest -version=CoreUnittest -main -preview=nosharedaccess -run import/core/atomic.d`